crypto/curve25519/curve25519.c

```diff
@@ -397,14 +395,18 @@ static void fe_invert(fe *out, const fe *z) {
 
 // return 0 if f == 0
 // return 1 if f != 0
-static int fe_isnonzero(const fe_loose *f) {
+static int fe_isnonzero_vartime(const fe_loose *f) {
   fe tight;
   fe_carry(&tight, f);
   uint8_t s[32];
   fe_tobytes(s, &tight);
 
-  static const uint8_t zero[32] = {0};
-  return CRYPTO_memcmp(s, zero, sizeof(zero)) != 0;
+  for (unsigned i = 0; i < 32; i++) {
```

Comment:
We use `size_t` for array indexes. I think we should use `sizeof(s)` instead of `32`.